test/correctness/debug_to_file_multiple_outputs.cpp

```diff
@@ -51,15 +51,17 @@ int main(int argc, char **argv) {
     assert(f && g && h);
 
     int header[5];
-    assert(fread((void *)(&header[0]), 4, 5, f) == 5);
```

Comment:
Might be good to pull the fix to the side-effecty asserts into a separate PR. I believe it's the only substantial code change (the rest is just messing with headers)?

Reply:
Yes. I can cherry pick those changes and move them to a separate PR.